tslearn/nonmyopic.py

```diff
@@ -0,0 +1,509 @@
+from sklearn.neural_network import MLPClassifier
+from sklearn.metrics import confusion_matrix
+from sklearn.base import BaseEstimator, ClassifierMixin
+from tslearn.utils import to_time_series_dataset
+from tslearn.clustering import TimeSeriesKMeans, silhouette_score
+import matplotlib.pyplot as plt
+import numpy as np
+import math
+from sklearn.metrics import roc_auc_score
+import os.path
+
+
+class NonMyopicEarlyClassification(BaseEstimator, ClassifierMixin):
+    """Early Classification modelling for time series using the model presented in [1]
+
+    Parameters
+    ----------
+    number_cluster : int
+        Number of clusters to form.
+
+    solver :
```

Comment:
Here, you stick to the original Dachraoui et al's version that rely on MLP classifier. Wouldn't it be more flexible, if you just had a `classifier` parameter that would take an estimator to be used for individual classifications?

Reply:
Then you could build (at fit time), the attribute `classifier_` by just cloning this one as many times as required (time length of `X`).